core/src/apps/cardano/sign_tx/multisig_signer.py

```diff
@@ -19,17 +19,26 @@ def __init__(
     ) -> None:
         super().__init__(ctx, msg, keychain)
 
+        self.signing_mode_title = "Confirming a multisig transaction."
+
     def _validate_tx_init(self) -> None:
         super()._validate_tx_init()
-        if (
-            self.msg.collateral_inputs_count != 0
-            or self.msg.required_signers_count != 0
-        ):
-            raise wire.ProcessError("Invalid tx signing request")
+        self._assert_tx_init_cond(self.msg.collateral_inputs_count == 0)
+        self._assert_tx_init_cond(self.msg.required_signers_count == 0)
+        self._assert_tx_init_cond(not self.msg.has_collateral_return)
+        self._assert_tx_init_cond(self.msg.total_collateral is None)
+        self._assert_tx_init_cond(self.msg.reference_inputs_count == 0)
+
+    def _should_show_tx_init(self) -> bool:
+        return True
 
-    async def _show_tx_init(self) -> None:
-        await layout.show_multisig_tx(self.ctx)
-        await super()._show_tx_init()
+    def _is_expert_view_allowed(self) -> bool:
+        return (
+            self.msg.has_output_details
+            or self.msg.withdrawals_count > 0
+            or self.msg.script_data_hash is not None
+            or self.msg.required_signers_count > 0
+        )
```

Comment:
Shouldn't we also add `or self.msg.has_auxiliary_data` here (and also in ordinary signing mode)?

Also, I consider this mechanism a bit dangerous - if we ever add new items that should be shown only in expert mode and forget to add the condition here, the nasty result will be that the user won't be able to choose expert mode for such tx (and view the item). But I'm not sure if we can do something about this.

Reply:
> Shouldn't we also add or self.msg.has_auxiliary_data here (and also in ordinary signing mode)?

Indeed, I somehow completely forgot about the auxiliary data here, thanks. Added in https://github.com/vacuumlabs/trezor-firmware/pull/79/commits/eddc71d2c1978a113e8acefe7ac1f7c77d413067.

> Also, I consider this mechanism a bit dangerous

As discussed offline - I'll give this another thought.

Reply:
Since the decision to always show the prompt has been made, this has become irrelevant.